### PR TITLE
rework: move features into dedicated module

### PR DESCRIPTION
**Pull request by tdevilleduc**
Signed-off-by: Thomas DEVILLE-DUC <thomas.deville-duc-prestataire@laposte.fr>

### CODE REVIEW

File: chaosk8s/deployment/probes.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    except urllib3.exceptions.ReadTimeoutError:
+        logger.debug("Timed out!")
+        return False
+
+
+def deployment_not_fully_available(name: str, ns: str = "default",
+                                   label_selector: str = "name in ({name})",
+                                   timeout: int = 30,
+                                   secrets: Secrets = None):
```
Comment (Lawouach) [inline chaosk8s/deployment/probes.py:108]: missing return type

File: chaosk8s/deployment/probes.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                "Deployment '{name}' is not healthy".format(name=name))
+
+    return True
+
+
+def _deployment_readiness_has_state(name: str, ready: bool,
+                                    ns: str = "default",
+                                    label_selector: str = "name in ({name})",
+                                    timeout: int = 30,
+                                    secrets: Secrets = None):
```
Comment (Lawouach) [inline chaosk8s/deployment/probes.py:63]: missing return type

File: chaosk8s/deployment/actions.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+from logzero import logger
+from kubernetes.client.rest import ApiException
+
+from chaosk8s import create_k8s_api_client
+
+__all__ = ["create_deployment", "delete_deployment"]
+
+
+def create_deployment(spec_path: str, ns: str = "default",
+                      secrets: Secrets = None):
```
Comment (Lawouach) [inline chaosk8s/deployment/actions.py:17]: missing return type

File: chaosk8s/deployment/actions.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            deployment = json.loads(f.read())
+        elif ext in ['.yml', '.yaml']:
+            deployment = yaml.load(f.read())
+        else:
+            raise ActivityFailed(
+                "cannot process {path}".format(path=spec_path))
+
+    v1 = client.AppsV1beta1Api(api)
+    resp = v1.create_namespaced_deployment(ns, body=deployment)
+    return resp
```
Comment (Lawouach) [inline chaosk8s/deployment/actions.py:36]: should we return the serialized version of the response?

usually the response is logged as-is in the chaostoolkit.log file and I'm not sure this format will be readable?

**Comment by Lawouach** [comment]
I'm happy to take it as-is once we have a changelog entry because it's a big change :)

**Comment by Lawouach** [comment]
Hey @rdrgmnzs, @enderv, I saw you folks pushed valuable PRs recently. This one from @tdevilleduc is useful because it reorganises the package a bit more cleanly. But I'm afraid it will mean your own PRs need a bit of work. I wanted to sync up with you before I merge any of them.

File: chaosk8s/deployment/actions.py (hunk 4 above)
Comment (tdevilleduc) [reply chaosk8s/deployment/actions.py]: I looked in other location, and create functions doesn't have return 
I've just deleted it

If returning response is wanted, we can create another PR by doing it everywhere

**Comment by enderv** [comment]
👍 once this is in I'll  merge it into my PR's and fix em up.

**Comment by rdrgmnzs** [comment]
Yep, same here. Once this lands I can re-base my PR.

**Comment by Lawouach** [comment]
Thank you all. Let's do this!

**Pull request merged**